src/Stepper/Stepper.js

```diff
@@ -0,0 +1,136 @@
+// @flow
+// @inheritedComponent Paper
+
+import React, { Children } from 'react';
+import type { Element, Node, ChildrenArray } from 'react';
+import classNames from 'classnames';
+import withStyles from '../styles/withStyles';
+import Paper from '../Paper';
+import StepConnector from './StepConnector';
+import Step from './Step';
+
+export const styles = (theme: Object) => ({
+  root: {
+    display: 'flex',
+    padding: theme.spacing.unit * 3,
+    // TODO: Guidelines state 24px, should we use `theme.spacing.unit * 3` or 24px?
+  },
+  horizontal: {
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+  vertical: {
+    flexDirection: 'column',
+  },
+});
+
+export type Orientation = 'horizontal' | 'vertical';
+
+type ProvidedProps = {
+  activeStep: number,
+  alternativeLabel: boolean,
+  classes: Object,
+  connector: Element<any>,
+  nonLinear: boolean,
+  orientation: Orientation,
+};
+
+export type Props = {
+  /**
+   * Set the active step (zero based index).
+   */
+  activeStep?: number,
+  /**
+   * If set to 'true' and orientation is horizontal,
+   * then the step label will be positioned under the icon.
+   */
+  alternativeLabel?: boolean,
+  /**
+   * Two or more `<Step />` components.
+   */
+  children: ChildrenArray<Element<typeof Step>>,
+  /**
+   * Useful to extend the style applied to components.
+   */
+  classes?: Object,
+  /**
+   * @ignore
+   */
+  className?: string,
+  /**
+   * A component to be placed between each step.
+   */
+  connector?: Element<typeof StepConnector> | Node,
+  /**
+   * If set the `Stepper` will not assist in controlling steps for linear flow
+   */
+  nonLinear?: boolean,
+  /**
+   * The stepper orientation (layout flow direction)
+   */
+  orientation?: Orientation,
+};
+
+function Stepper(props: ProvidedProps & Props) {
+  const {
+    activeStep,
+    alternativeLabel,
+    classes,
+    className: classNameProp,
+    children,
+    connector: connectorProp,
+    nonLinear,
+    orientation,
+    ...other
+  } = props;
+
+  const className = classNames(
+    classes.root,
+    classNameProp,
+    alternativeLabel ? null : classes[orientation],
+  );
+
+  const connector = connectorProp ? React.cloneElement(connectorProp, { orientation }) : null;
+  const numChildren = Children.count(children);
```

Comment:
As far as I know, it's more performant to use `toArray` once than having children count and children map.

Reply:
Yep, it's certainly faster, even with the extra `React.cloneElement` that is required - https://jsperf.com/react-children-count-map-vs-toarray-count-map

Fixed in b14200e6f